github/commitstatus.go

```diff
@@ -244,15 +254,23 @@ func checkForRetry(res httpResponse, waitTime, maxSleepTime time.Duration) time.
 	// If you exceed the rate limit, the response will have a 403 status and the x-ratelimit-remaining header will be 0
 	// https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
 	case res.statusCode == http.StatusForbidden && res.rateLimitRemaining == 0:
-		sleepTime := time.Until(res.rateLimitReset)
+		sleepTime := res.rateLimitReset.Sub(res.date) + jitter
 		if sleepTime > maxSleepTime {
 			return 0
 		} else {
-			return sleepTime
+			// Don't allow negative sleep times. Minimum sleep time should be equal to jitter time.
```

Comment:
The comment is somewhat misleading:
We avoid negative sleep time only by using the server's notion of clock, the jitter is unrelated, but the comment is about the max, which is unrelated to negative values, because sleepTime is guaranteed to be non negative.

So I would remove the comment entirely.

Reply:
See the comment above on how to make it clearer what is happening

Reply:
Fixed as suggested in the comment below.